src/relax/backend/vm/codegen_vm.cc

```diff
@@ -514,6 +522,39 @@ class CodeGenVM : public ExprFunctor<Instruction::Arg(const Expr&)> {
     return ret;
   }
 
+  Instruction::Arg EmitGetCapturedCUDAGraph(const CallNode* call) {
+    auto gv = Downcast<GlobalVar>(call->args[0]);
+    auto func_name = gv->name_hint;
+    TVMRetValue func_name_constant;
+    func_name_constant = func_name;
+    auto func_name_index = builder_->EmitConstant(func_name_constant);
+    size_t dst_register = NewRegister();
+
+    builder_->EmitCall("vm.builtin.get_captured_cuda_graph",
+                       {Instruction::Arg(Instruction::kRegister, Instruction::kVMRegister),
+                        Instruction::Arg(Instruction::kConstIdx, func_name_index)},
+                       dst_register);
+    return Instruction::Arg(Instruction::kRegister, dst_register);
+  }
+
+  Instruction::Arg EmitCUDAGraphBeginCapture(const CallNode* call) {
+    size_t dst_register = NewRegister();
+    builder_->EmitCall("vm.builtin.cuda_graph_begin_capture", {}, dst_register);
+    return Instruction::Arg(Instruction::kRegister, dst_register);
+  }
+
+  Instruction::Arg EmitCUDAGraphEndCapture(const CallNode* call) {
+    size_t dst_register = NewRegister();
+    builder_->EmitCall("vm.builtin.cuda_graph_end_capture", {}, dst_register);
+    return Instruction::Arg(Instruction::kRegister, dst_register);
+  }
+
+  Instruction::Arg EmitCUDAGraphLaunch(const CallNode* call) {
+    size_t dst_register = NewRegister();
+    builder_->EmitCall("vm.builtin.cuda_graph_launch", ConvertArgs(GetRef<Call>(call)),
+                       dst_register);
```

Comment:
NOTE after we bring back let us move to https://github.com/tlc-pack/relax/blob/relax/src/relax/backend/vm/vm_builtin_lower.cc instead